docs/guides/model_selection.md

```diff
@@ -0,0 +1,254 @@
+Models:
+├── Directly Modified:
+│   └── sushi.items
+└── Indirectly Modified:
+    ├── sushi.order_items
+    ├── sushi.customer_revenue_by_day
+    ├── sushi.waiter_revenue_by_day
+    ├── sushi.customer_revenue_lifetime
+    └── sushi.top_waiters
+```
+
+#### Select `items+`
+
+If we specify the `--select-model` option with the downstream `+` to select `"sushi.items+"`, the `sushi.order_items` model is selected and classified as directly modified because it is downstream of `sushi.items`:
+
+```bash
+❯ sqlmesh plan dev --select-model "sushi.items+"
+Summary of differences against `dev`:
+Models:
+├── Directly Modified:
+│   ├── sushi.items
+│   └── sushi.order_items
+└── Indirectly Modified:
+    ├── sushi.waiter_revenue_by_day
+    ├── sushi.customer_revenue_lifetime
+    ├── sushi.customer_revenue_by_day
+    └── sushi.top_waiters
+```
+
+#### Select `*items`
+
+If we specify the `--select-model` option with the wildcard `*` to select `"sushi.*items"`, both `sushi.items` and `sushi.order_items` are selected because they match the wildcard:
+
+```bash
+❯ sqlmesh plan dev --select-model "sushi.*items"
+Summary of differences against `dev`:
+Models:
+├── Directly Modified:
+│   ├── sushi.order_items
+│   └── sushi.items
+└── Indirectly Modified:
+    ├── sushi.waiter_revenue_by_day
+    ├── sushi.top_waiters
+    ├── sushi.customer_revenue_by_day
+    └── sushi.customer_revenue_lifetime
+```
+
+### Backfill examples
+
+#### No backfill selection
+
+Recall that a plan with no selection or backfill options includes all four models, two of which were directly and two of which were indirectly modified.
+
+The `--backfill-model` option does not affect whether a model is included in a plan (i.e., it will still appear in the output shown in the selection examples above). Instead, it is excluded from the list of models needing backfill at the bottom of the plan's output.
+
+With no options specified, the `plan` will backfill all six models. The backfills occur in the `sushi__dev` schema because we are creating a plan for the `dev` environment:
+
+```bash hl_lines="16 17 21 24"
+❯ sqlmesh plan dev
+Summary of differences against `dev`:
+Models:
+├── Directly Modified:
+│   ├── sushi.items
+│   └── sushi.order_items
+└── Indirectly Modified:
+    ├── sushi.customer_revenue_by_day
+    ├── sushi.customer_revenue_lifetime
+    ├── sushi.waiter_revenue_by_day
+    └── sushi.top_waiters
+
+< output omitted>
+
+Models needing backfill (missing dates):
+├── sushi__dev.items: 2023-12-01 - 2023-12-07
+├── sushi__dev.order_items: 2023-12-01 - 2023-12-07
+├── sushi__dev.customer_revenue_by_day: 2023-12-01 - 2023-12-07
+├── sushi__dev.customer_revenue_lifetime: 2023-12-01 - 2023-12-07
+├── sushi__dev.waiter_revenue_by_day: 2023-12-01 - 2023-12-07
+└── sushi__dev.top_waiters: 2023-12-01 - 2023-12-07
+```
+
+#### Backfill `sushi.waiter_revenue_by_day`
+
+If we specify the `--backfill-model` option with `"sushi.waiter_revenue_by_day"`, two things happen:
```

Comment:
We should clarify that the reason why `items` and `order_items` have been included is because they are upstream from `waiter_revenue_by_day` and `--backfill-model` **always** includes upstream regardless of whether a leading `+` was specified in the selection expression.